src/main/java/com/exadel/discount/repository/CouponRepository.java

```diff
@@ -37,4 +33,8 @@
     @EntityGraph(attributePaths = {"discount", "discount.category", "discount.vendorLocations", "discount.tags",
             "discount.vendor", "discount.vendorLocations.city", "discount.vendorLocations.city.country"})
     List<Coupon> findAllByIdIn(Iterable<UUID> ids, Sort sort);
+
+    @EntityGraph(attributePaths = {"user", "discount"})
+    Optional<Coupon> findByDiscountIdAndAndUserEmail(@Param("discountId") UUID discountId,
+                                                     @Param("userEmail") String userEmail);
```

Comment:
You don't need @Param if you don't use @Query

Reply:
Done